tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc

```diff
@@ -1643,27 +1643,34 @@ class OpConverterTest : public ::testing::Test {
 
   // Helper method to run both validation and conversion, and check the output
   // shape.
-  void RunValidationAndConversion(const NodeDef& node_def, const Status& status,
-                                  const char* output_name,
-                                  const std::vector<int>& exp_out_dims) {
+  void RunValidationAndConversion(
+      const NodeDef& node_def, const Status& status, const char* output_name,
+      const std::vector<std::vector<int>>& exp_out_dims) {
     RunValidationAndConversion(node_def, status.code(),
                                status.error_message().c_str(), true);
     if (status.ok()) {
-      TRT_TensorOrWeights output;
-      TF_EXPECT_OK(GetTensorOrWeights(output_name, &output));
-      ASSERT_TRUE(output.is_tensor());
-      if (converter_->use_implicit_batch() && !exp_out_dims.empty()) {
-        // We only check output shape implicit batch mode. In dynamic shape
-        // mode we need to wait for the concrate input shapes to be defined
-        // (by setBindingDimensions before enqueue) before we can check
-        // whether the output dims are equal.
-        //
-        // TODO(tamas) enable this check in explicit_batch_mode
-
-        // Removing batch dim
-        auto out_dims =
-            std::vector<int>(exp_out_dims.begin() + 1, exp_out_dims.end());
-        ExpectTrtDimsEqualsArray(out_dims, output.tensor()->getDimensions());
+      if (converter_->use_implicit_batch()) {
+        for (int i = 0; i < exp_out_dims.size(); i++) {
+          TRT_TensorOrWeights output;
+          string name = i == 0 ? output_name : StrCat(output_name, ":", i);
+          TF_EXPECT_OK(GetTensorOrWeights(name.c_str(), &output));
+          ASSERT_TRUE(output.is_tensor());
+          if (!exp_out_dims[i].empty()) {
+            // We only check output shape implicit batch mode. In dynamic shape
+            // mode we need to wait for the concrate input shapes to be defined
+            // (by setBindingDimensions before enqueue) before we can check
+            // whether the output dims are equal.
+            //
+            // TODO(tfeher): Enable this check in explicit_batch_mode.
```

Comment:
Let's move the TODO to right before the if(use_implicit_batch). The comment here also implies that the check for output shapes for dynamic shape can't happen here, instead in a later stage. Can we modify the comment to explicitly state so?

We check the output shapes here https://github.com/sourcecode369/tensorflow-1/blob/9f446aba8aaeb2b3c4c6e5ba1ab4cf31494b8a64/tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc#L1806-L1811
So that is not the only check we need?